rebasehelper/helpers/download_helper.py

```diff
@@ -0,0 +1,182 @@
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+# Authors: Petr Hracek <phracek@redhat.com>
+#          Tomas Hozza <thozza@redhat.com>
+
+import os
+import sys
+import time
+import urllib
+
+import requests
+import six
+
+from six.moves import urllib
+
+from rebasehelper.exceptions import DownloadError
+from rebasehelper.logger import logger
+
+
+class DownloadHelper(object):
+
+    """Class for downloading files and getting Response objects."""
+
+    @staticmethod
+    def progress(download_total, downloaded, start_time):
+        """Prints download progress and expected remaining time of the download directly to the standard output.
+
+        Args:
+            download_total (int, float): Size of the file which is being downloaded.
+            downloaded (int, float): Size of the part of the already downloaded part of the file.
+            start_time (float): Time when the download started.
+
+        """
+        bar_width = 32
+        infinite_step = 256 * 1024  # move every 256 kilobytes
+
+        delta = time.time() - start_time
+
+        def format_time(t):
+            h, rem = divmod(int(t), 3600)
+            m, s = divmod(rem, 60)
+            return '{:0>2d}:{:0>2d}:{:0>2d}'.format(h, m, s)
+
+        def format_size(s):
+            units = [' ', 'K', 'M', 'G', 'T']
+            i = 0
+            while s >= 1024.0 and i < len(units) - 1:
+                s /= 1024.0
+                i += 1
+            return '{:>7.2F}{}'.format(s, units[i])
+
+        if download_total < 0:
+            # infinite progress bar
+            pct = ' ' * 4
+            pos = int(downloaded / infinite_step) % (bar_width - 5)
+            bar = '[{}<=>{}]'.format(' ' * pos, ' ' * (bar_width - 5 - pos))
+            ts = ' in {}'.format(format_time(delta))
+        else:
+            r = float(downloaded) / float(download_total) if download_total else 0.0
+            pct = '{:>3d}%'.format(int(r * 100))
+            pos = int(r * (bar_width - 3))
+            bar = '[{}>{}]'.format('=' * pos, ' ' * (bar_width - 3 - pos))
+            ts = 'eta {}'.format(format_time(delta / r - delta) if r > 0.0 else ' ' * 7 + '?')
+
+        size = format_size(downloaded)
+
+        # no point to log progress, write directly to stdout
+        sys.stdout.write('\r{}{}  {}  {} '.format(pct, bar, size, ts))
+        sys.stdout.flush()
+
+    @staticmethod
+    def request(url, **kwargs):
+        """Gets a Reponse object from HTTP, HTTPS or FTP URL.
```

Comment:
There is a typo in `Response`, but I'd change the whole description to something like:
`Performs a HTTP request or a FTP RETR command.`.